src/viking.js

```diff
@@ -1,11 +1,77 @@
 // Soldier
-class Soldier {}
+class Soldier {
+  constructor(health, strength) {
+    this.health = health;
+    this.strength = strength;
+  }
+  attack() {
+    return this.strength;
+  }
+  receiveDamage(damage) {
+    this.health -= damage;
+  }
+}
 
 // Viking
-class Viking {}
+class Viking extends Soldier {
+  constructor(name, health, strength) {
+    super(health, strength);
+    this.name = name;
+  }
+  receiveDamage(damage) {
+    this.health -= damage;
+    if (this.health === 0) {
+      return `${this.name} has died in act of combat`;
+    } else {
+      return `${this.name} has received ${damage} points of damage`;
+    }
+  }
+  battleCry() {
+    return "Odin Owns You All!";
+  }
+}
 
 // Saxon
-class Saxon {}
+class Saxon extends Soldier {
+    constructor(health, strength, name) {
```

Comment:
Saxon will not receive a name so there's no need for the constructor neither the super, since it's inheriting everything from the Soldier class:

```
class Saxon extends Soldier {
  receiveDamage(damage) {
    this.health -= damage;

    if (this.health <= 0) {
      return `A Saxon has died in combat`;
    } else {
      return `A Saxon has received ${damage} points of damage`;
    }
  }
}
```